File: nahomy-c2-backend/src/infrastructure/config/DatabaseConfig.ts
import { DataSource } from 'typeorm'
import {
  UsuarioEntity,
  ProcesoIntencionEntity,
  UsuarioIntencionEntity,
  FuncionPrealmacenadaEntity,
  EjecucionAutomatizadaEntity,
  FeedbackReporteEntity,
  LogAuditoriaEntity,
  MetricaRendimientoEntity,
} from '../persistence/typeorm/entities'

export function createDataSource(): DataSource {
  return new DataSource({
    type: 'postgres',
    host: process.env.DB_HOST || 'localhost',
    port: Number(process.env.DB_PORT) || 5432,
    username: process.env.DB_USER || 'nahomy',
    password: process.env.DB_PASSWORD || 'nahomy_secret',
    database: process.env.DB_NAME || 'nahomy_c2',
    entities: [
      UsuarioEntity,
      ProcesoIntencionEntity,
      UsuarioIntencionEntity,
      FuncionPrealmacenadaEntity,
      EjecucionAutomatizadaEntity,
      FeedbackReporteEntity,
      LogAuditoriaEntity,
      MetricaRendimientoEntity,
    ],
    synchronize: false,
    logging: process.env.NODE_ENV === 'development' ? true : false,
    ssl: process.env.DB_SSL === 'true' ? { rejectUnauthorized: false } : false,
  })
}

File: packages/server/src/infrastructure/config/database_config.py
import os


def get_database_url() -> str:
    user = os.getenv("DB_USER", "nahomy")
    password = os.getenv("DB_PASSWORD", "nahomy_secret")
    host = os.getenv("DB_HOST", "localhost")
    port = os.getenv("DB_PORT", "5432")
    name = os.getenv("DB_NAME", "nahomy_c2")
    return f"postgresql+asyncpg://{user}:{password}@{host}:{port}/{name}"
